source/common/router/upstream_request.h

```diff
@@ -203,7 +204,8 @@ class UpstreamRequest : public Logger::Loggable<Logger::Id::router>,
   const MonotonicTime start_time_;
   // This is wrapped in an optional, since we want to avoid computing zero size headers when in
   // reality we just didn't get a response back.
-  absl::optional<uint64_t> response_headers_size_{};
+  absl::optional<uint64_t> response_headers_size_;
+  absl::optional<size_t> response_headers_count_;
```

Comment:
It doesn't matter much, but will it be possible to use uint32_t instead of size_t?

Reply:
I left as-is. It seemed weird to me to lose precision when converting to a `uint32_t` only to convert back to a `uint64_t` when recording the histogram value. not that it really matters in practice and i don't feel that strongly about it. if you think i should still change it, just let me know.